assets/js/blocks/attribute-filter/block.js

```diff
@@ -0,0 +1,194 @@
+			return {};
+		}
+		// If doing an "AND" query, we need to remove current taxonomy query so counts are not affected.
+		const modifiedQueryState =
+			'and' === queryType
+				? productAttributes.filter(
+						( item ) => item.attribute !== taxonomy
+				  )
+				: productAttributes;
+
+		// Take current query and remove paging args.
+		return {
+			...queryState,
+			orderby: undefined,
+			order: undefined,
+			per_page: undefined,
+			page: undefined,
+			attributes: modifiedQueryState,
+			calculate_attribute_counts: [ taxonomy ],
+		};
+	}, [ queryState, taxonomy, queryType, productAttributes ] );
+
+	const { results: allTerms, isLoading: allTermsIsLoading } = useCollection( {
+		namespace: '/wc/blocks',
+		resourceName: 'products/attributes/terms',
+		resourceValues: [ attributeId ],
+		query: { per_page: 100 },
+	} );
+
+	const {
+		results: filteredCounts,
+		isLoading: filteredCountsIsLoading,
+	} = useCollection( {
+		namespace: '/wc/store',
+		resourceName: 'products/collection-data',
+		query: {
+			...countsQueryState,
+		},
+	} );
+
+	const getLabel = useCallback(
+		( name, count ) => {
+			return (
+				<Fragment>
+					{ name }
+					{ showCounts && count && (
+						<span className="wc-block-attribute-filter-list-count">
+							{ count }
+						</span>
+					) }
+				</Fragment>
+			);
+		},
+		[ showCounts ]
+	);
+
+	const getFilteredTerm = useCallback(
+		( id ) => {
+			if ( ! filteredCounts.attribute_counts ) {
+				return {};
+			}
+			return filteredCounts.attribute_counts.find(
+				( { term } ) => term === id
+			);
+		},
+		[ filteredCounts ]
+	);
+
+	/**
+	 * Compare intersection of all terms and filtered counts to get a list of options to display.
+	 */
+	useEffect( () => {
+		if ( filteredCountsIsLoading || allTermsIsLoading ) {
+			return;
+		}
+
+		const newOptions = [];
+
+		allTerms.forEach( ( term ) => {
+			const filteredTerm = getFilteredTerm( term.id );
+
+			if ( ! filteredTerm && filteredCounts !== null ) {
+				return;
+			}
+
+			newOptions.push( {
+				key: term.slug,
+				label: getLabel(
+					term.name,
+					filteredTerm ? filteredTerm.count : term.count
+				),
+				checked: false,
+			} );
+		} );
+
+		setOptions( newOptions );
+	}, [
+		filteredCounts,
+		allTerms,
+		showCounts,
+		filteredCountsIsLoading,
+		allTermsIsLoading,
+		getFilteredTerm,
+		getLabel,
+	] );
+
+	useEffect( () => {
+		const newProductAttributes = productAttributes.filter(
+			( item ) => item.attribute !== taxonomy
+		);
+		const slug = checkedOptions.join( ',' );
+
+		if ( slug ) {
+			const updatedQuery = {
+				attribute: taxonomy,
+				operator: 'or' === queryType ? 'in' : 'and',
+				slug,
+			};
+			newProductAttributes.push( updatedQuery );
+		}
+
+		setProductAttributes( sortBy( newProductAttributes, 'attribute' ) );
+	}, [ checkedOptions, taxonomy, productAttributes ] );
+
+	const onChange = useCallback( ( checked ) => {
+		setCheckedOptions( checked );
+	}, [] );
+
+	if ( ! taxonomy ) {
+		return null;
+	}
```

Comment:
Could we move this check closer to the top of the functional component? If we are returning `null`, I guess it's better to return as early as possible to avoid useless computation.

Reply:
Conditionals must be after hooks.